playbooks/roles/ocp-upgrade/tasks/eus_upgrades.yaml

```diff
@@ -0,0 +1,196 @@
+---
+
+- name: Pause the worker's MachineConfigPool
+  kubernetes.core.k8s:
+    state: present
+    definition:
+      kind: MachineConfigPool
+      metadata:
+        name: worker
+      spec: 
+        paused: true
+
+- name: Patch the ClusterVersion with the eus intermediate upgrade channel
+  kubernetes.core.k8s:
+    state: present
+    definition:
+      kind: ClusterVersion
+      metadata:
+        name: version
+      spec: 
+        channel: "{{ eus_upgrade_channel }}"
+        upstream: "{{ upstream }}"
+  when: eus_upgrade_channel != ""
+
+- name: Get the OCP version 
+  kubernetes.core.k8s_info:
+    kind: ClusterVersion
+    name: version
+  register: ocp_version
+
+- name: Providing the administrator acknowledgement for 4.8 to 4.9 upgrades
+  kubernetes.core.k8s:
+    state: present
+    definition:
+      kind: ConfigMap
+      metadata:
+        name: admin-acks
+        namespace: openshift-config
+      data: 
+        ack-4.8-kube-1.22-api-removals-in-4.9: "true"
+  when: ocp_version.resources[0].status.history[0].version | regex_search('4.8') == "4.8"
+
+- name: Upgrade cluster with upgrade version
+  shell: oc adm upgrade --to {{ eus_intermediate_upgrade_version }}
+  when: eus_intermediate_upgrade_image == "" and eus_intermediate_upgrade_version != ""
+
+- name: Upgrade cluster with intermediate upgrade image
+  shell: |
+    oc adm upgrade --force --to-image="{{ eus_intermediate_upgrade_image }}" --allow-explicit-upgrade
+  when: eus_intermediate_upgrade_image != ""
+
+- name: Upgrade the cluster to the latest available build in the channel
+  shell: oc adm upgrade --to-latest
+  when: 
+  - eus_upgrade_channel != "" and eus_intermediate_upgrade_image == ""
+  - eus_intermediate_upgrade_version == ""
+
+- name: Waiting until upgrade completes
+  pause:
+    minutes: "{{ pause_time }}"
+
+- name: Check for upgrade completion
+  shell: oc get clusterversion -o json|jq ".items[0].status.history[0].state"
+  register: state
+  until: state.stdout | replace('\"', '') == "Completed"
+  retries: 5
+  delay: "{{ delay_time }}"
+
+- name: Check all co are in 'Available' state and not in 'Progressing' or 'Degraded' state
```

Comment:
Can we have a simple reusable role for polling clusteroperator and other checks which are repeated?

Reply:
yes, I will create an role for it.